orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/SaveServiceAccountTask.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.front50.tasks;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.fiat.model.UserPermission;
+import com.netflix.spinnaker.fiat.model.resources.Role;
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import com.netflix.spinnaker.fiat.shared.FiatStatus;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.FiatServiceAccount;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Save a pipeline-scoped Fiat Service Account. The roles from
+ * this service account are used for authorization decisions when
+ * the pipeline is executed from an automated trigger.
+ */
+
+@Component
+@Slf4j
+public class SaveServiceAccountTask implements RetryableTask {
+
+  private static final String SERVICE_ACCOUNT_SUFFIX = "@managed-service-account";
+
+  @Autowired(required = false)
+  private FiatStatus fiatStatus;
+
+  @Autowired(required = false)
+  private Front50Service front50Service;
+
+  @Autowired(required = false)
+  private FiatPermissionEvaluator fiatPermissionEvaluator;
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.SECONDS.toMillis(1);
+  }
+
+  @Override
+  public long getTimeout() {
+    return TimeUnit.SECONDS.toMillis(30);
+  }
+
+  @Nonnull
+  @SuppressWarnings("unchecked")
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (!fiatStatus.isEnabled()) {
+      throw new UnsupportedOperationException("Fiat is not enabled, cannot save permissions.");
+    }
+
+    if (front50Service == null) {
```

Comment:
My best estimation is that there should never be a time when `front50Service` is null, so you can probably remove this check & the `required=false` annotation above.

Reply:
The only place I see `front50.enabled` set to false is in `halconfig/orca-bootstrap.yml` 
Would requiring `front50Service` always be a problem for halyard bootstrap? @lwander

Reply:
Yes it would break the bootstrapping process. Orca doesn't need front50 when only running executions.